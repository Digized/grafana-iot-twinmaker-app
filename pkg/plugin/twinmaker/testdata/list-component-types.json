{
    "ComponentTypeSummaries": [
        {
            "Arn": "arn:aws:iottwinmaker:us-east-1:166800769179:workspace/AlarmWorkspace/component-type/com.example.cookiefactory.alarm",
            "ComponentTypeId": "com.example.cookiefactory.alarm",
            "CreationDateTime": "2022-04-27T17:29:43.035Z",
            "Description": null,
            "Status": {
                "Error": {
                    "Code": null,
                    "Message": null
                },
                "State": "ACTIVE"
            },
            "UpdateDateTime": "2022-04-27T17:29:43.035Z"
        },
        {
            "Arn": "arn:aws:iottwinmaker:us-east-1:166800769179:workspace/AlarmWorkspace/component-type/com.example.cookiefactory.mixer",
            "ComponentTypeId": "com.example.cookiefactory.mixer",
            "CreationDateTime": "2022-04-27T17:29:43.217Z",
            "Description": null,
            "Status": {
                "Error": {
                    "Code": null,
                    "Message": null
                },
                "State": "ACTIVE"
            },
            "UpdateDateTime": "2022-04-27T17:29:43.217Z"
        },
        {
            "Arn": "arn:aws:iottwinmaker:us-east-1:166800769179:workspace/AlarmWorkspace/component-type/com.example.cookiefactory.space",
            "ComponentTypeId": "com.example.cookiefactory.space",
            "CreationDateTime": "2022-04-27T17:29:43.539Z",
            "Description": null,
            "Status": {
                "Error": {
                    "Code": null,
                    "Message": null
                },
                "State": "ACTIVE"
            },
            "UpdateDateTime": "2022-04-27T17:29:43.539Z"
        },
        {
            "Arn": "arn:aws:iottwinmaker:us-east-1:166800769179:workspace/AlarmWorkspace/component-type/com.example.cookiefactory.watertank",
            "ComponentTypeId": "com.example.cookiefactory.watertank",
            "CreationDateTime": "2022-04-27T17:29:43.377Z",
            "Description": null,
            "Status": {
                "Error": {
                    "Code": null,
                    "Message": null
                },
                "State": "ACTIVE"
            },
            "UpdateDateTime": "2022-04-27T17:29:43.377Z"
        },
        {
            "Arn": "arn:aws:iottwinmaker:us-east-1:166800769179:workspace/AlarmWorkspace/component-type/com.example.timestream-telemetry",
            "ComponentTypeId": "com.example.timestream-telemetry",
            "CreationDateTime": "2022-04-27T17:29:42.887Z",
            "Description": null,
            "Status": {
                "Error": {
                    "Code": null,
                    "Message": null
                },
                "State": "ACTIVE"
            },
            "UpdateDateTime": "2022-04-27T17:29:42.887Z"
        },
        {
            "Arn": "arn:aws:iottwinmaker:us-east-1:iotrociaccount:workspace/AmazonOwnedTypesWorkspace/component-type/com.amazon.iotsitewise.connector",
            "ComponentTypeId": "com.amazon.iotsitewise.connector",
            "CreationDateTime": "2021-11-13T00:25:32.467Z",
            "Description": "IoT SiteWise connector for syncing asset properties",
            "Status": {
                "Error": {
                    "Code": null,
                    "Message": null
                },
                "State": "ACTIVE"
            },
            "UpdateDateTime": "2021-11-13T00:25:32.467Z"
        },
        {
            "Arn": "arn:aws:iottwinmaker:us-east-1:iotrociaccount:workspace/AmazonOwnedTypesWorkspace/component-type/com.amazon.iotsitewise.connector.edgevideo",
            "ComponentTypeId": "com.amazon.iotsitewise.connector.edgevideo",
            "CreationDateTime": "2021-11-13T00:25:34.069Z",
            "Description": "IoT SiteWise connector for syncing edgevideo asset properties",
            "Status": {
                "Error": {
                    "Code": null,
                    "Message": null
                },
                "State": "ACTIVE"
            },
            "UpdateDateTime": "2021-11-13T00:25:34.069Z"
        },
        {
            "Arn": "arn:aws:iottwinmaker:us-east-1:iotrociaccount:workspace/AmazonOwnedTypesWorkspace/component-type/com.amazon.iottwinmaker.alarm.basic",
            "ComponentTypeId": "com.amazon.iottwinmaker.alarm.basic",
            "CreationDateTime": "2021-11-13T00:25:35.967Z",
            "Description": "Abstract alarm component type",
            "Status": {
                "Error": {
                    "Code": null,
                    "Message": null
                },
                "State": "ACTIVE"
            },
            "UpdateDateTime": "2021-11-13T00:25:35.967Z"
        },
        {
            "Arn": "arn:aws:iottwinmaker:us-east-1:iotrociaccount:workspace/AmazonOwnedTypesWorkspace/component-type/com.amazon.iottwinmaker.documents",
            "ComponentTypeId": "com.amazon.iottwinmaker.documents",
            "CreationDateTime": "2021-11-13T00:25:30.797Z",
            "Description": "Component type for storing document urls",
            "Status": {
                "Error": {
                    "Code": null,
                    "Message": null
                },
                "State": "ACTIVE"
            },
            "UpdateDateTime": "2021-11-13T00:25:30.797Z"
        },
        {
            "Arn": "arn:aws:iottwinmaker:us-east-1:iotrociaccount:workspace/AmazonOwnedTypesWorkspace/component-type/com.amazon.iottwinmaker.parameters",
            "ComponentTypeId": "com.amazon.iottwinmaker.parameters",
            "CreationDateTime": "2021-11-13T00:25:38.067Z",
            "Description": "Component type for storing parameter values",
            "Status": {
                "Error": {
                    "Code": null,
                    "Message": null
                },
                "State": "ACTIVE"
            },
            "UpdateDateTime": "2021-11-13T00:25:38.067Z"
        }
    ],
    "MaxResults": null,
    "NextToken": null,
    "WorkspaceId": "AlarmWorkspace"
}