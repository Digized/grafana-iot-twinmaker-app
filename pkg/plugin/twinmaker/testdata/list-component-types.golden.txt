ðŸŒŸ This was machine generated.  Do not edit. ðŸŒŸ

Frame[0] {
    "custom": {}
}
Name: 
Dimensions: 4 Fields by 10 Rows
+--------------------------------------------+---------------------------------------------------------------+-----------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------+
| Name: componentId                          | Name: description                                             | Name: created                     | Name: arn                                                                                                                                   |
| Labels:                                    | Labels:                                                       | Labels:                           | Labels:                                                                                                                                     |
| Type: []*string                            | Type: []*string                                               | Type: []time.Time                 | Type: []*string                                                                                                                             |
+--------------------------------------------+---------------------------------------------------------------+-----------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------+
| com.example.cookiefactory.alarm            | null                                                          | 2022-04-27 17:29:43.035 +0000 UTC | arn:aws:iottwinmaker:us-east-1:166800769179:workspace/AlarmWorkspace/component-type/com.example.cookiefactory.alarm                         |
| com.example.cookiefactory.mixer            | null                                                          | 2022-04-27 17:29:43.217 +0000 UTC | arn:aws:iottwinmaker:us-east-1:166800769179:workspace/AlarmWorkspace/component-type/com.example.cookiefactory.mixer                         |
| com.example.cookiefactory.space            | null                                                          | 2022-04-27 17:29:43.539 +0000 UTC | arn:aws:iottwinmaker:us-east-1:166800769179:workspace/AlarmWorkspace/component-type/com.example.cookiefactory.space                         |
| com.example.cookiefactory.watertank        | null                                                          | 2022-04-27 17:29:43.377 +0000 UTC | arn:aws:iottwinmaker:us-east-1:166800769179:workspace/AlarmWorkspace/component-type/com.example.cookiefactory.watertank                     |
| com.example.timestream-telemetry           | null                                                          | 2022-04-27 17:29:42.887 +0000 UTC | arn:aws:iottwinmaker:us-east-1:166800769179:workspace/AlarmWorkspace/component-type/com.example.timestream-telemetry                        |
| com.amazon.iotsitewise.connector           | IoT SiteWise connector for syncing asset properties           | 2021-11-13 00:25:32.467 +0000 UTC | arn:aws:iottwinmaker:us-east-1:iotrociaccount:workspace/AmazonOwnedTypesWorkspace/component-type/com.amazon.iotsitewise.connector           |
| com.amazon.iotsitewise.connector.edgevideo | IoT SiteWise connector for syncing edgevideo asset properties | 2021-11-13 00:25:34.069 +0000 UTC | arn:aws:iottwinmaker:us-east-1:iotrociaccount:workspace/AmazonOwnedTypesWorkspace/component-type/com.amazon.iotsitewise.connector.edgevideo |
| com.amazon.iottwinmaker.alarm.basic        | Abstract alarm component type                                 | 2021-11-13 00:25:35.967 +0000 UTC | arn:aws:iottwinmaker:us-east-1:iotrociaccount:workspace/AmazonOwnedTypesWorkspace/component-type/com.amazon.iottwinmaker.alarm.basic        |
| com.amazon.iottwinmaker.documents          | Component type for storing document urls                      | 2021-11-13 00:25:30.797 +0000 UTC | arn:aws:iottwinmaker:us-east-1:iotrociaccount:workspace/AmazonOwnedTypesWorkspace/component-type/com.amazon.iottwinmaker.documents          |
| com.amazon.iottwinmaker.parameters         | Component type for storing parameter values                   | 2021-11-13 00:25:38.067 +0000 UTC | arn:aws:iottwinmaker:us-east-1:iotrociaccount:workspace/AmazonOwnedTypesWorkspace/component-type/com.amazon.iottwinmaker.parameters         |
+--------------------------------------------+---------------------------------------------------------------+-----------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------+


====== TEST DATA RESPONSE (arrow base64) ======
FRAME=QVJST1cxAAD/////cAIAABAAAAAAAAoADgAMAAsABAAKAAAAFAAAAAAAAAEEAAoADAAAAAgABAAKAAAACAAAAIAAAAADAAAATAAAACgAAAAEAAAAJP7//wgAAAAMAAAAAAAAAAAAAAAFAAAAcmVmSWQAAABE/v//CAAAAAwAAAAAAAAAAAAAAAQAAABuYW1lAAAAAGT+//8IAAAAGAAAAA0AAAB7ImN1c3RvbSI6e319AAAABAAAAG1ldGEAAAAABAAAAEwBAADUAAAAbAAAAAQAAADW/v//FAAAADgAAAA4AAAAAAAFATQAAAABAAAABAAAAMT+//8IAAAADAAAAAMAAABhcm4ABAAAAG5hbWUAAAAAAAAAALT+//8DAAAAYXJuAAAAEgAYABQAAAATAAwAAAAIAAQAEgAAABQAAAA8AAAARAAAAAAAAApEAAAAAQAAAAQAAAAo////CAAAABAAAAAHAAAAY3JlYXRlZAAEAAAAbmFtZQAAAAAAAAAAAAAGAAgABgAGAAAAAAADAAcAAABjcmVhdGVkAJ7///8UAAAAQAAAAEAAAAAAAAUBPAAAAAEAAAAEAAAAjP///wgAAAAUAAAACwAAAGRlc2NyaXB0aW9uAAQAAABuYW1lAAAAAAAAAACE////CwAAAGRlc2NyaXB0aW9uAAAAEgAYABQAEwASAAwAAAAIAAQAEgAAABQAAABIAAAATAAAAAAABQFIAAAAAQAAAAwAAAAIAAwACAAEAAgAAAAIAAAAFAAAAAsAAABjb21wb25lbnRJZAAEAAAAbmFtZQAAAAAAAAAABAAEAAQAAAALAAAAY29tcG9uZW50SWQAAAAAAP////9IAQAAFAAAAAAAAAAMABYAFAATAAwABAAMAAAA+AcAAAAAAAAUAAAAAAAAAwQACgAYAAwACAAEAAoAAAAUAAAAyAAAAAoAAAAAAAAAAAAAAAsAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAsAAAAAAAAADAAAAAAAAAAUAEAAAAAAACAAQAAAAAAAAIAAAAAAAAAiAEAAAAAAAAsAAAAAAAAALgBAAAAAAAA4AAAAAAAAACYAgAAAAAAAAAAAAAAAAAAmAIAAAAAAABQAAAAAAAAAOgCAAAAAAAAAAAAAAAAAADoAgAAAAAAACwAAAAAAAAAGAMAAAAAAADZBAAAAAAAAAAAAAAEAAAACgAAAAAAAAAAAAAAAAAAAAoAAAAAAAAABQAAAAAAAAAKAAAAAAAAAAAAAAAAAAAACgAAAAAAAAAAAAAAAAAAAAAAAAAfAAAAPgAAAF0AAACAAAAAoAAAAMAAAADqAAAADQEAAC4BAABQAQAAAAAAAGNvbS5leGFtcGxlLmNvb2tpZWZhY3RvcnkuYWxhcm1jb20uZXhhbXBsZS5jb29raWVmYWN0b3J5Lm1peGVyY29tLmV4YW1wbGUuY29va2llZmFjdG9yeS5zcGFjZWNvbS5leGFtcGxlLmNvb2tpZWZhY3Rvcnkud2F0ZXJ0YW5rY29tLmV4YW1wbGUudGltZXN0cmVhbS10ZWxlbWV0cnljb20uYW1hem9uLmlvdHNpdGV3aXNlLmNvbm5lY3RvcmNvbS5hbWF6b24uaW90c2l0ZXdpc2UuY29ubmVjdG9yLmVkZ2V2aWRlb2NvbS5hbWF6b24uaW90dHdpbm1ha2VyLmFsYXJtLmJhc2ljY29tLmFtYXpvbi5pb3R0d2lubWFrZXIuZG9jdW1lbnRzY29tLmFtYXpvbi5pb3R0d2lubWFrZXIucGFyYW1ldGVyc+ADAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMwAAAHAAAACNAAAAtQAAAOAAAAAAAAAASW9UIFNpdGVXaXNlIGNvbm5lY3RvciBmb3Igc3luY2luZyBhc3NldCBwcm9wZXJ0aWVzSW9UIFNpdGVXaXNlIGNvbm5lY3RvciBmb3Igc3luY2luZyBlZGdldmlkZW8gYXNzZXQgcHJvcGVydGllc0Fic3RyYWN0IGFsYXJtIGNvbXBvbmVudCB0eXBlQ29tcG9uZW50IHR5cGUgZm9yIHN0b3JpbmcgZG9jdW1lbnQgdXJsc0NvbXBvbmVudCB0eXBlIGZvciBzdG9yaW5nIHBhcmFtZXRlciB2YWx1ZXPAlFjfCtHpFkCuMeoK0ekWwAJj/QrR6RZAFrvzCtHpFsBHhtYK0ekWwLImpGXzthZAR6MDZvO2FsB1xHRm87YWQIWcQGXzthbA6u/xZvO2FgAAAABzAAAA5gAAAFkBAADQAQAARAIAAMUCAABQAwAA1AMAAFYEAADZBAAAAAAAAGFybjphd3M6aW90dHdpbm1ha2VyOnVzLWVhc3QtMToxNjY4MDA3NjkxNzk6d29ya3NwYWNlL0FsYXJtV29ya3NwYWNlL2NvbXBvbmVudC10eXBlL2NvbS5leGFtcGxlLmNvb2tpZWZhY3RvcnkuYWxhcm1hcm46YXdzOmlvdHR3aW5tYWtlcjp1cy1lYXN0LTE6MTY2ODAwNzY5MTc5OndvcmtzcGFjZS9BbGFybVdvcmtzcGFjZS9jb21wb25lbnQtdHlwZS9jb20uZXhhbXBsZS5jb29raWVmYWN0b3J5Lm1peGVyYXJuOmF3czppb3R0d2lubWFrZXI6dXMtZWFzdC0xOjE2NjgwMDc2OTE3OTp3b3Jrc3BhY2UvQWxhcm1Xb3Jrc3BhY2UvY29tcG9uZW50LXR5cGUvY29tLmV4YW1wbGUuY29va2llZmFjdG9yeS5zcGFjZWFybjphd3M6aW90dHdpbm1ha2VyOnVzLWVhc3QtMToxNjY4MDA3NjkxNzk6d29ya3NwYWNlL0FsYXJtV29ya3NwYWNlL2NvbXBvbmVudC10eXBlL2NvbS5leGFtcGxlLmNvb2tpZWZhY3Rvcnkud2F0ZXJ0YW5rYXJuOmF3czppb3R0d2lubWFrZXI6dXMtZWFzdC0xOjE2NjgwMDc2OTE3OTp3b3Jrc3BhY2UvQWxhcm1Xb3Jrc3BhY2UvY29tcG9uZW50LXR5cGUvY29tLmV4YW1wbGUudGltZXN0cmVhbS10ZWxlbWV0cnlhcm46YXdzOmlvdHR3aW5tYWtlcjp1cy1lYXN0LTE6aW90cm9jaWFjY291bnQ6d29ya3NwYWNlL0FtYXpvbk93bmVkVHlwZXNXb3Jrc3BhY2UvY29tcG9uZW50LXR5cGUvY29tLmFtYXpvbi5pb3RzaXRld2lzZS5jb25uZWN0b3Jhcm46YXdzOmlvdHR3aW5tYWtlcjp1cy1lYXN0LTE6aW90cm9jaWFjY291bnQ6d29ya3NwYWNlL0FtYXpvbk93bmVkVHlwZXNXb3Jrc3BhY2UvY29tcG9uZW50LXR5cGUvY29tLmFtYXpvbi5pb3RzaXRld2lzZS5jb25uZWN0b3IuZWRnZXZpZGVvYXJuOmF3czppb3R0d2lubWFrZXI6dXMtZWFzdC0xOmlvdHJvY2lhY2NvdW50OndvcmtzcGFjZS9BbWF6b25Pd25lZFR5cGVzV29ya3NwYWNlL2NvbXBvbmVudC10eXBlL2NvbS5hbWF6b24uaW90dHdpbm1ha2VyLmFsYXJtLmJhc2ljYXJuOmF3czppb3R0d2lubWFrZXI6dXMtZWFzdC0xOmlvdHJvY2lhY2NvdW50OndvcmtzcGFjZS9BbWF6b25Pd25lZFR5cGVzV29ya3NwYWNlL2NvbXBvbmVudC10eXBlL2NvbS5hbWF6b24uaW90dHdpbm1ha2VyLmRvY3VtZW50c2Fybjphd3M6aW90dHdpbm1ha2VyOnVzLWVhc3QtMTppb3Ryb2NpYWNjb3VudDp3b3Jrc3BhY2UvQW1hem9uT3duZWRUeXBlc1dvcmtzcGFjZS9jb21wb25lbnQtdHlwZS9jb20uYW1hem9uLmlvdHR3aW5tYWtlci5wYXJhbWV0ZXJzAAAAAAAAABAAAAAMABQAEgAMAAgABAAMAAAAEAAAACwAAAA4AAAAAAAEAAEAAACAAgAAAAAAAFABAAAAAAAA+AcAAAAAAAAAAAAAAAAAAAAACgAMAAAACAAEAAoAAAAIAAAAgAAAAAMAAABMAAAAKAAAAAQAAAAk/v//CAAAAAwAAAAAAAAAAAAAAAUAAAByZWZJZAAAAET+//8IAAAADAAAAAAAAAAAAAAABAAAAG5hbWUAAAAAZP7//wgAAAAYAAAADQAAAHsiY3VzdG9tIjp7fX0AAAAEAAAAbWV0YQAAAAAEAAAATAEAANQAAABsAAAABAAAANb+//8UAAAAOAAAADgAAAAAAAUBNAAAAAEAAAAEAAAAxP7//wgAAAAMAAAAAwAAAGFybgAEAAAAbmFtZQAAAAAAAAAAtP7//wMAAABhcm4AAAASABgAFAAAABMADAAAAAgABAASAAAAFAAAADwAAABEAAAAAAAACkQAAAABAAAABAAAACj///8IAAAAEAAAAAcAAABjcmVhdGVkAAQAAABuYW1lAAAAAAAAAAAAAAYACAAGAAYAAAAAAAMABwAAAGNyZWF0ZWQAnv///xQAAABAAAAAQAAAAAAABQE8AAAAAQAAAAQAAACM////CAAAABQAAAALAAAAZGVzY3JpcHRpb24ABAAAAG5hbWUAAAAAAAAAAIT///8LAAAAZGVzY3JpcHRpb24AAAASABgAFAATABIADAAAAAgABAASAAAAFAAAAEgAAABMAAAAAAAFAUgAAAABAAAADAAAAAgADAAIAAQACAAAAAgAAAAUAAAACwAAAGNvbXBvbmVudElkAAQAAABuYW1lAAAAAAAAAAAEAAQABAAAAAsAAABjb21wb25lbnRJZACYAgAAQVJST1cx
